service/storage_service.cc

```diff
@@ -3108,10 +3108,10 @@ future<> storage_service::replicate_to_all_cores(mutable_token_metadata_ptr tmpt
             }
         });
         // Prepare per-table erms.
-        co_await container().invoke_on_all([&] (storage_service& ss) {
+        co_await container().invoke_on_all([&] (storage_service& ss) -> future<> {
             auto& db = ss._db.local();
             auto tmptr = pending_token_metadata_ptr[this_shard_id()];
-            db.get_tables_metadata().for_each_table([&] (table_id id, lw_shared_ptr<replica::table> table) {
+            co_await db.get_tables_metadata().for_each_table_gently([&] (table_id id, lw_shared_ptr<replica::table> table) {
```

Comment:
An improvement, but doesn't solve the problem if there is a dominant table.

Reply:
What do you mean by dominant table, and why would it present a problem?

As far as I could see `tablet_aware_replication_strategy::do_make_replication_map` has constant time complexity.  Did I miss anything?

Reply:
Ah, right. I was confused, in PR https://github.com/scylladb/scylladb/pull/21289 we make erm construction O(tablet count), but that will add its own preemption points.